xds/internal/xdsclient/loadreport.go

```diff
@@ -28,7 +28,9 @@ import (
 // It returns a Store for the user to report loads, a function to cancel the
 // load reporting stream.
 func (c *clientImpl) ReportLoad(server *bootstrap.ServerConfig) (*load.Store, func()) {
+	c.authorityMu.Lock()
 	a, err := c.newAuthority(server)
```

Comment:
Should we name the function `newAuthorityLocked` to better indicate you should have the lock when calling it?  I see it's in the doc comments but those obviously weren't sufficient in this case.

Reply:
Done. Rename this and several other methods.